zerver/views/onboarding_steps.py

```diff
@@ -1,23 +1,60 @@
+from datetime import timedelta
+
+from django.conf import settings
 from django.http import HttpRequest, HttpResponse
+from django.utils.timezone import now as timezone_now
 from django.utils.translation import gettext as _
+from pydantic import Json
 
 from zerver.actions.onboarding_steps import do_mark_onboarding_step_as_read
+from zerver.actions.scheduled_messages import check_schedule_message
 from zerver.decorator import human_users_only
 from zerver.lib.exceptions import JsonableError
 from zerver.lib.onboarding_steps import ALL_ONBOARDING_STEPS
+from zerver.lib.request import RequestNotes
 from zerver.lib.response import json_success
 from zerver.lib.typed_endpoint import typed_endpoint
 from zerver.models import UserProfile
+from zerver.models.users import get_system_bot
 
 
 @human_users_only
 @typed_endpoint
 def mark_onboarding_step_as_read(
-    request: HttpRequest, user: UserProfile, *, onboarding_step: str
+    request: HttpRequest,
+    user: UserProfile,
+    *,
+    onboarding_step: str,
+    schedule_onboarding_video_reminder_delay: Json[int] | None = None,
 ) -> HttpResponse:
     if not any(step.name == onboarding_step for step in ALL_ONBOARDING_STEPS):
         raise JsonableError(
             _("Unknown onboarding_step: {onboarding_step}").format(onboarding_step=onboarding_step)
         )
+
+    if schedule_onboarding_video_reminder_delay is not None:
+        assert onboarding_step == "onboarding_video"
+
+        realm = user.realm
+        sender = get_system_bot(settings.WELCOME_BOT, realm.id)
+        client = RequestNotes.get_notes(request).client
```

Comment:
Do we have other system bots messages use this as a client? `_internal_prep_message` uses "Internal" as the client, which seems more correct.

Reply:
Umm, No. But I thought this goes in the category of "message scheduled by the user on behalf of welcome bot" -- so client should be noted accordingly.

But we can use "Internal" too -- some user action resulted in internal bot message which was not instant but scheduled.

Changed it to Internal.